File: cpp/66.plus-one.cpp
#include <vector>
using std::vector;

class Solution {
public:
  vector<int> plusOne(vector<int> &digits) {
    int carry = 1;
    int tmp;

    for (int n = digits.size() - 1; n >= 0 && carry; n--) {
      tmp = carry + digits[n];
      carry = tmp / 10;
      digits[n] = tmp % 10;
    }

    if (carry) {
      digits.insert(digits.begin(), 1);
    }

    return digits;
  }
};

File: golang/66.plus-one.go

func plusOne(digits []int) []int {
	var tmp int
	carry := 1

	for n := len(digits) - 1; n >= 0 && carry > 0; n-- {
		tmp = carry + digits[n]
		carry = tmp / 10
		digits[n] = tmp % 10
	}

	if carry > 0 {
		digits = append([]int{1}, digits...)
	}

	return digits
}
